src/rust/src/rustzcash.rs

```diff
@@ -1340,3 +1340,37 @@ pub extern "system" fn librustzcash_mmr_hash_node(
 
     0
 }
+
+#[no_mangle]
+pub extern "system" fn librustzcash_zebra_crypto_sign_verify_detached(
+    sig: *const [u8; 64],
+    m: *const u8,
+    mlen: u64,
+    pk: *const [u8; 32],
+) -> isize {
+    use ed25519_zebra::{Signature, VerificationKey};
+    use std::convert::TryFrom;
+
+    let sig = Signature::from(*unsafe {
+        match sig.as_ref() {
+            Some(sig) => sig,
+            None => return 1,
```

Comment:
We are emulating libsodium's API here, so it would be helpful to define constants that make clear these error values are taken from libsodium, and don't match how we return errors in the other FFIs:

```suggestion
            None => return LIBSODIUM_ERROR,
```

and then just above this function:
```rust
const LIBSODIUM_ERROR: isize = 1;
```